indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/PemEncodable.kt

```diff
@@ -0,0 +1,110 @@
+package at.asitplus.signum.indispensable.asn1
+
+import at.asitplus.KmmResult
+import at.asitplus.catching
+import kotlin.io.encoding.Base64
+import kotlin.io.encoding.ExperimentalEncodingApi
+
+/**
+ * Specialization of [Asn1Encodable], able to produce PEM-encoded strings
+ * as per [RFC 1421](https://datatracker.ietf.org/doc/html/rfc1421#section-4.4).
+ * Use in tandem with [PemDecodable].
+ */
+interface PemEncodable<A : Asn1Element> : Asn1Encodable<A> {
+
+    /**
+     * Encapsulation boundary string.
+     * Will be automatically fenced.
+     * Make sure it matches the corresponding [PemDecodable.ebString]!
+     */
+    val ebString: String
+
+    /**
+     * To be implemented for custom encoding of PEM-encoded payload.
+     * Defaults to [encodeToDer]
+     */
+    @Throws(Throwable::class)
+    fun binaryEncodePayload(): ByteArray = encodeToDer()
+}
+
+/**
+ * Specialization of [Asn1Decodable], able to parse PEM-encoded strings
+ * as per [RFC 1421](https://datatracker.ietf.org/doc/html/rfc1421).
+ * Use in tandem with [PemEncodable].
+ */
+interface PemDecodable<A : Asn1Element, T : PemEncodable<A>> : Asn1Decodable<A, T> {
+
+    /**
+     * Encapsulation boundary string.
+     * Will be automatically fenced.
+     * Make sure it matches the corresponding [PemEncodable.ebString]!
+     */
+    val ebString: String
+
+    /**
+     * To be implemented for custom decoding of PEM-encoded payload.
+     * Defaults to [decodeFromDer] and validates pre- and postEb
+     */
+    @Throws(Throwable::class)
+    fun binaryDecodePayload(ebString: String, src: ByteArray): T {
+        require(ebString == this.ebString) { "PEM encoded EB string does not match with ${this.ebString}: $ebString" }
+        return decodeFromDer(src)
+    }
+}
+
+
+private const val FENCE_PRE = "-----BEGIN "
+private const val FENCE_POST = "-----END "
+private const val FENCE_AFTER = "-----"
+
+/**-----BEGIN [PemEncodable.ebString]-----*/
+val PemEncodable<*>.preEB: String get() = "$FENCE_PRE$ebString$FENCE_AFTER"
```

Comment:
i would prefer the more expressive `${this.ebString}`

Reply:
which other ebString could it be?